### PR TITLE
Adding the hashed key as the subject to the JWT

### PR DESCRIPTION
**Pull request by thomasm-ttd**
Not enforced yet.

### CODE REVIEW

File: src/main/java/com/uid2/shared/Utils.java [hunk 1]
```diff
@@ -191,4 +193,11 @@ public static InputStream convertHttpResponseToInputStream(HttpResponse<String>
 
         return new ByteArrayInputStream(responseBytes);
     }
+    public static MessageDigest createMessageDigest() {
```
Comment (vishalegbert-ttd) [inline src/main/java/com/uid2/shared/Utils.java:196]: Maybe rename this function to make it specific to SHA-512?

File: src/main/java/com/uid2/shared/attest/JwtValidationResponse.java [hunk 2]
```diff
@@ -28,9 +29,11 @@ public JwtValidationResponse withRoles(Role... roles) {
         return this;
     }
     public JwtValidationResponse withRoles(String roles) {
-        String[] parts = roles.split(",");
-        var t = Arrays.stream(parts).map(r -> Role.valueOf(r)).collect(Collectors.toList());
-        this.roles = Set.copyOf(t);
+        if (roles != null && !roles.isBlank()) {
+            String[] parts = roles.split(",");
+            var t = Arrays.stream(parts).map(r -> Role.valueOf(r)).collect(Collectors.toList());
+            this.roles = Set.copyOf(t);
```
Comment (vishalegbert-ttd) [inline src/main/java/com/uid2/shared/attest/JwtValidationResponse.java:35]: Should it default to empty?

Comment (thomasm-ttd) [reply src/main/java/com/uid2/shared/attest/JwtValidationResponse.java:35]: Renamed

File: src/main/java/com/uid2/shared/Utils.java (hunk 1 above)
Comment (thomasm-ttd) [reply src/main/java/com/uid2/shared/Utils.java:196]: Renamed

File: src/main/java/com/uid2/shared/attest/JwtValidationResponse.java (hunk 2 above)
Comment (thomasm-ttd) [reply src/main/java/com/uid2/shared/attest/JwtValidationResponse.java:35]: Done

**Review by vishalegbert-ttd — APPROVED** [review]

**Pull request merged**